components/Carbon.js

```diff
@@ -402,27 +409,31 @@ function selectedLinesReducer({ prevLine, selected }, { type, lineNumber, numLin
   }
 }
 
-function useGutterClickHandler(props) {
-  const editorRef = React.useRef(null)
+function useSelectedLines(props, editorRef) {
   const [state, dispatch] = React.useReducer(selectedLinesReducer, {
     prevLine: null,
     selected: {}
   })
```

Comment:
@nemesv quicks thoughts. Now that we are also story `selectedLines` in the `Editor` state, we can probably remove this local state and just call something like `this.props.setSelectedLines` instead. That way we don't have this state in two places. 

We _don't_ need it in this PR, but do you think that sounds feasible?